tests/sentry/incidents/test_logic.py

```diff
@@ -1425,6 +1534,119 @@ def test_pagerduty_not_existing(self):
+            },
+        )
+        action = create_alert_rule_trigger_action(
+            self.trigger,
+            type,
+            target_type,
+            target_identifier=channel_id,
+            integration_id=integration.id,
+        )
+        assert action.alert_rule_trigger == self.trigger
+        assert action.type == type.value
+        assert action.target_type == target_type.value
+        assert action.target_display == channel_id
+        assert action.integration_id == integration.id
+
+    @responses.activate
+    def test_discord_no_integration(self):
+        base_url: str = "https://discord.com/api/v10"
+        channel_id = "channel-id"
+        guild_id = "example-discord-server"
+        guild_name = "Server Name"
+        type = AlertRuleTriggerAction.Type.DISCORD
+        target_type = AlertRuleTriggerAction.TargetType.SPECIFIC
+        responses.add(
+            method=responses.GET,
+            url=f"{base_url}/channels/{channel_id}",
+            json={
+                "guild_id": f"{guild_id}",
+                "name": f"{guild_name}",
+            },
+        )
+
+        with pytest.raises(InvalidTriggerActionError):
+            create_alert_rule_trigger_action(
+                self.trigger,
+                type,
+                target_type,
+                target_identifier=channel_id,
+                integration_id=123,
+            )
+
+    @responses.activate
+    def test_discord_invalid_channel(self):
+        base_url: str = "https://discord.com/api/v10"
+        channel_id = "channel-id"
+        guild_id = "example-discord-server"
+        guild_name = "Server Name"
+        integration = Integration.objects.create(
+            provider="discord",
+            name="Example Discord",
+            external_id=f"{guild_id}",
+            metadata={
+                "guild_id": f"{guild_id}",
+                "name": f"{guild_name}",
+            },
+        )
+        type = AlertRuleTriggerAction.Type.DISCORD
+        target_type = AlertRuleTriggerAction.TargetType.SPECIFIC
+        responses.add(method=responses.GET, url=f"{base_url}/channels/{channel_id}", status=400)
+
+        with pytest.raises(InvalidTriggerActionError):
+            create_alert_rule_trigger_action(
+                self.trigger,
+                type,
+                target_type,
+                target_identifier=channel_id,
+                integration_id=integration.id,
+            )
+
+    def test_discord_not_responding(self):
+        integration = Integration.objects.create(
+            provider="discord",
+            name="Example Discord",
+            external_id="guild-id",
+            metadata={
+                "guild_id": "guild_id",
+                "name": "guild_name",
+            },
+        )
+        type = AlertRuleTriggerAction.Type.DISCORD
+        target_type = AlertRuleTriggerAction.TargetType.SPECIFIC
+        with pytest.raises(IntegrationError):
```

Comment:
Is this test supposed to mimic a timeout from Discord? Maybe it could be mocked similar to [this](https://github.com/getsentry/sentry/blob/master/tests/sentry/integrations/test_client.py#L185) test.

Reply:
I don't think this test is needed since it is meant for the get_channel_id type function in create_alert_rule_tigger_action (which is not needed since we are not getting channel id by name)